### PR TITLE
Enable a portable report directory

### PR DESCRIPTION
**Pull request by standage**
MicroHapulator generates numerous data files, analysis results, and statistics to facilitate interpretation. The space required to store these files can reach into the dozens of gigabytes for a sufficiently large run.

In this PR, I've modified the workflow and the working directory structure to enable a practical and more easily-shared report directory. In short:

- the `report.html` and `marker-details-report.html` files are now written to a new dedicated `report/` directory
- all `.png` image files are copied to this directory as a final step of the Snakemake workflow
- Bootstrap CSS and Jquery JavaScript files are now bundled with MicroHapulator and copied to the `report/` directory so it will render correctly even without an internet connection

To share, just zip the `report/` file and send!

----------

- [x] Changes are clearly described above
- [x] Any relevant issue threads are referenced in the description
- [x] Any new features are tested (see the [development manual](https://microhapulator.readthedocs.io/en/stable/devel.html) for details)
- [x] CLI documentation (see [docs/cli.md](docs/cli.md)) and Python API documentation (see [microhapulator/api.py](microhapulator/api.py)) are up-to-date and in sync
- [x] Substantial changes are documented in CHANGELOG.md (see https://keepachangelog.com/en/1.0.0/)

### CODE REVIEW

File: microhapulator/api.py [hunk 1]
```diff
@@ -680,7 +680,7 @@ def plot_haplotype_calls(result, outdir, sample=None, plot_marker_name=True, ign
             count_dict = {allele: count for allele, count in count_dict.items() if count >= static}
         counts = count_dict.values()
         alleles = count_dict.keys()
-        plt.figure(figsize=(4, 4), dpi=150)
+        plt.figure(figsize=(4, 4), dpi=100)
```
Comment (standage) [inline microhapulator/api.py:683]: Reduces the size of the report directory a bit.

File: microhapulator/data/template.html [hunk 2]
```diff
@@ -101,8 +101,10 @@ <h2>Table of Contents</h2>
                 {% endblock %}
             </ol>
             <p>
-                A summary of the statistics presented in this report is aggregated in a single table available at <code>analysis/summary.tsv</code> in the working directory.
-                Full-resolution graphics for each figure are also available in each <code>analysis/{samplename}/</code> subdirectory.
+                The data and statistics from which this report is populated are available in their entirety in the MicroHapulator working directory.
+                Per-sample results and full-resolution graphics are available in each <code>analysis/{samplename}</code> subdirectory in the working directory.
+                The entire working directory is large and not convenient for sharing.
+                Accordingly, graphics, JavaScript, and CSS assets are copied to the <code>report/</code> directory, which can then be compressed in a ZIP archive that should be better suited for e.g. email transfer.
```
Comment (standage) [inline microhapulator/data/template.html:107]: Updated this outdated description and suggested how the report can be shared.

**Review by rnmitchell — APPROVED** [review]
This looks good to me! The separate report folder works great.

**Review by standage — COMMENTED** [review]
Whoops, forgot to publish these pending comments. Thanks for the review!

**Pull request merged**